core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidator.java

```diff
@@ -228,6 +245,21 @@ public void validateScalingAdjustment(String instanceGroupName, Integer scalingA
         }
     }
 
+    public void validateInstanceStatusesForStart(Stack stack, InstanceGroupAdjustmentV4Request instanceGroupAdjustmentJson) {
+        if (upscaleEvent(instanceGroupAdjustmentJson.getScalingAdjustment())) {
+            List<InstanceMetaData> instanceMetaDataList =
+                    stack.getInstanceMetaDataAsList()
+                            .stream()
+                            .filter(im -> im.getInstanceGroupName().equalsIgnoreCase(instanceGroupAdjustmentJson.getInstanceGroup()) && im.isStopped())
+                            .collect(Collectors.toList());
+            if (instanceMetaDataList.isEmpty() || instanceGroupAdjustmentJson.getScalingAdjustment() > instanceMetaDataList.size()) {
```

Comment:
I don't think we should put in this check. It is acceptable to request an upscale for more nodes than are currently available.
Also, eventually we'd want to handle instances where the CM health for the nodes is in a state from which we can recover (i.e. Instance RUNNING, CM decommissioned / services STOPPED)

Reply:
Hmm, it will create array index out of bound exception here,

`instancesToStart = stoppedInstancesInCbHg.subList(0, numInstancesToStart);`

Reply:
Fixing this in the start improvements PR. Thanks for pointing it out. Tests, if they were part of the PR, would've identified this. Anyway, will add those in the follow-up jira created.